backend/graminsta/core/serializers.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: UTF-8 -*-
+
+"""
+Here defines model serializers.
+"""
+
+from django.contrib.auth import models as auth_models
+from rest_framework import serializers
+from .models import UserInfo
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """
+    UserSerializer defined for User model.
+    """
+    class Meta:
+        """
+        Meta Information
+        """
+        model = auth_models.User
+        fields = (
+            'username',
+            'first_name',
+            'last_name',
+            'email',
+            'password',
+        )
+
+
+class UserInfoSerializer(serializers.ModelSerializer):
+    """
+    UserInfoSerializer for  UserInfo class
+    """
+    user = UserSerializer(required=True)
+
+    class Meta:
+        """
+        Meta Information
+        """
+        model = UserInfo
+        fields = (
+            'user',
+            'age',
+            'gender',
+        )
+
+    def create(self, validated_data):
+        """
+        Overriding the default create method of the Model Serializer.
+        :param validated_data: data containing all the details of student
+        :return: returns a successfully created Userinfo record
+        """
+        user_data = validated_data.pop('user')
+        user = UserSerializer.create(
+            UserSerializer(), validated_data=user_data)
```

Comment:
Usually, when creating a `UserInfo` object, the `user` field should be only a primary key field (`int` in this case), instead of providing data for `User` and create it in the `UserInfo` creation process.

That is, we will need another abstraction like `services.py` which provides services for `views.py` to use, in some function of the `services.py`, it will take in the `User` data and `UserInfo` data and create `User` object then `UserInfo` object in transaction manner.

Reply:
In One-To-One relationship, `user` field can be `UserSerializer`, which is much easier to serialize and deserialize. 
I have implemented the version that uses the primary key to create UserInfo, but it is very inconvenient to deserialize `UserInfo` object.